### PR TITLE
Add listen for errors sample.

### PR DESCRIPTION
**Pull request by noerog**

### CODE REVIEW

File: pubsub/cloud-client/subscriber.py [hunk 1]
```diff
@@ -110,6 +110,34 @@ def callback(message):
...
+    """Receives messages and catches errors from a pull subscription."""
+    subscriber = pubsub_v1.SubscriberClient()
+    subscription_path = subscriber.subscription_path(
+        project, subscription_name)
+
+    def callback(message):
+        try:
+            print('Received message: {}'.format(message))
+            message.ack()
+            future.result()
```
Comment (jonparrott) [inline pubsub/cloud-client/subscriber.py]: You shouldreplace lines 136-138 with the `try: future.result() except: ...`.

File: pubsub/cloud-client/subscriber.py [hunk 2]
```diff
@@ -110,6 +110,31 @@ def callback(message):
...
+    # Blocks the thread while messages are coming in through the stream. Any
+    # exceptions that crop up on the thread will be set on the future.
+    future = subscription.open(callback)
+    try:
+        future.result()
+    except Exception as e:
+        print(
+            'Listening for messages on {} threw an Exception: {}.'.format(
+                subscription_name, e))
+        subscription.close()
```
Comment (jonparrott) [inline pubsub/cloud-client/subscriber.py]: I don't think you *have* to call `subscription.close()`, I'm pretty sure it happens automatically.

File: pubsub/cloud-client/subscriber.py [hunk 3]
```diff
@@ -110,6 +110,31 @@ def callback(message):
...
+    raise
```
Comment (jonparrott) [inline pubsub/cloud-client/subscriber.py]: This needs to be indented into the `except` block

File: pubsub/cloud-client/subscriber.py (hunk 2 above)
Comment (noerog) [reply pubsub/cloud-client/subscriber.py]: I'm fine with removing it. Thanks.

File: pubsub/cloud-client/subscriber.py (hunk 3 above)
Comment (noerog) [reply pubsub/cloud-client/subscriber.py]: Done, thanks.

**Comment by jonparrott** [comment]
Thanks, @noerog!

**Pull request merged**